src/libcore/slice/mod.rs

```diff
@@ -5859,9 +5859,6 @@ where
         if self.len() != other.len() {
```

Comment:
(15 lines above) There's still a pointer comparison in the `default fn equal`, probably remove in both places, if the comparisons should be removed